go/database/mpt/live_trie_test.go

```diff
@@ -190,6 +191,100 @@ func TestLiveTrie_Fail_Read_Data(t *testing.T) {
 	}
 }
 
+func TestLiveTrie_VisitAccount(t *testing.T) {
+	for _, config := range allMptConfigs {
+		t.Run(config.Name, func(t *testing.T) {
+			trie, err := OpenFileLiveTrie(t.TempDir(), config, NodeCacheConfig{Capacity: 16384})
+			if err != nil {
+				t.Fatalf("failed to open empty archive: %v", err)
+			}
+			defer func() {
+				if err := trie.Close(); err != nil {
+					t.Errorf("failed to close archive: %v", err)
+				}
+			}()
+
+			const (
+				Addresses = 125
+			)
+
+			// there are no accounts at the beginning
+			for i := 0; i < Addresses; i++ {
+				addr := common.AddressFromNumber(i)
+				if err := trie.VisitAccount(addr, MakeVisitor(func(node Node, _ NodeInfo) VisitResponse {
+					t.Errorf("unexpected node: %v", node)
+					return VisitResponseContinue
+				})); err != nil {
+					t.Fatalf("failed to visit account: %v", err)
+				}
+			}
+
+			// insert addresses only
+			for i := 0; i < Addresses; i++ {
+				if err := trie.SetAccountInfo(common.AddressFromNumber(i), AccountInfo{Nonce: common.Nonce{1}}); err != nil {
+					t.Fatalf("failed to set account info: %v", err)
+				}
+			}
+
+			// there are no keys in the accounts
+			for i := 0; i < Addresses; i++ {
+				addr := common.AddressFromNumber(i)
+				if err := trie.VisitAccount(addr, MakeVisitor(func(node Node, _ NodeInfo) VisitResponse {
```

Comment:
This part should somehow check that an account is indeed visited.

Reply:
added